MST/src/CMakeLists.txt

```diff
@@ -0,0 +1,140 @@
+target_sources(mst_main
```

Comment:
You should add all files except `mst2.F90` to `mst_lib`. The executable target `mst_main` should only contain `mst2.F90`.